lib/module.js

```diff
@@ -489,7 +489,9 @@ Module._initPaths = function() {
 
   var nodePath = process.env['NODE_PATH'];
   if (nodePath) {
-    paths = nodePath.split(path.delimiter).concat(paths);
+    paths = nodePath.split(path.delimiter).concat(paths).filter(function(path) {
+      return !!path
+    });
```

Comment:
No need to filter `paths` here, its entries aren't user-supplied. A small perf gain would be:

````js
paths = nodePath.split(path.delimiter).filter(function(path) {
  return !!path
}).concat(paths);
````